src/Modules/UngradedModules.jl

```diff
@@ -180,6 +180,22 @@ function (==)(F::FreeMod, G::FreeMod)
   return F.R == G.R && rank(F) == rank(G) && F.S == G.S
 end
 
+# the two methods below are needed for the implementation of is_surjective
+function (==)(G::SubquoModule, F::FreeMod)
+  return F == G
+end
+
+function (==)(F::FreeMod, G::SubquoModule)
+  base_ring(F) === base_ring(G) || return false
+  ambient_free_module(G) === F || return false
+  if isdefined(G, :quo) 
+    iszero(G.quo) || return false
+  end
+  all(e -> e in G, gens(F)) || return false
+  return true
+end
```

Comment:
@jankoboehm : Without these methods `is_surjective` would fail for homomorphisms with a `FreeMod` as codomain. Do you agree with adding these?

Reply:
Fine.